README.md
# FLUTTER PROFILE PICTURE

 Automatically generate profile picture with random first name and background color. But you can still provide pictures if you have them. As the default color, based on the name of the first letter.

[![Flutter Test](https://github.com/adityadees/flutter_profile_picture/actions/workflows/flutter-test.yml/badge.svg)](https://github.com/adityadees/flutter_profile_picture/actions/workflows/flutter-test.yml)
[![Build GitHub Pages](https://github.com/adityadees/flutter_profile_picture/actions/workflows/gh-pages.yml/badge.svg)](https://github.com/adityadees/flutter_profile_picture/actions/workflows/gh-pages.yml)
[![License: MIT](https://img.shields.io/badge/License-MIT-green.svg)](https://opensource.org/licenses/MIT)
[![pub package](https://img.shields.io/pub/v/flutter_profile_picture.svg)](https://pub.dartlang.org/packages/flutter_profile_picture)
[![GitHub popularity](https://img.shields.io/github/stars/adityadees/flutter_profile_picture?logo=github&logoColor=white)](https://github.com/adityadees/flutter_profile_picture/stargazers/)
![version](https://img.shields.io/github/v/release/adityadees/flutter_profile_picture)

![bandicam 2021-09-22 20-35-48-079 (online-video-cutter com)](https://user-images.githubusercontent.com/37553901/134374902-7247ec0d-8c00-4218-99c1-701cc1f01565.gif)

## Style

### Default Style

![image](https://user-images.githubusercontent.com/37553901/134119014-3b51bea5-e0bf-49b1-aa3e-8d5f91c6e92d.png)

``` dart
ProfilePicture(
    name: 'Dees',
    radius: 31,
    fontsize: 21,
);

ProfilePicture(
    name: 'Aditya Dharmawan Saputra',
    radius: 31,
    fontsize: 21,
);
```

### Random Color

![image](https://user-images.githubusercontent.com/37553901/134119324-8271b270-c695-4141-b079-bdc1c09ec0b0.png)

``` dart
ProfilePicture(
    name: 'Aditya Dharmawan Saputra',
    radius: 31,
    fontsize: 21,
    random: true,
);
```

### Max letter

![image](https://user-images.githubusercontent.com/37553901/134119854-72deba5c-67e5-42cd-b227-8dcf2b5beb9f.png)

``` dart

ProfilePicture(
    name: 'Aditya Dharmawan Saputra',
    radius: 31,
    fontsize: 21,
    count: 3,
);
```

### Empty Name / Blank

![image](https://user-images.githubusercontent.com/37553901/134119842-3a56f038-516e-4e08-a884-27eac8db73eb.png)

``` dart
ProfilePicture(
    name: '',
    radius: 31,
    fontsize: 21,
);
```

### With Images

![image](https://user-images.githubusercontent.com/37553901/134119814-8199fe68-cbc7-4a8d-8d4f-4abb16343a80.png)

``` dart
ProfilePicture(
    name: 'Aditya Dharmawan Saputra',
    radius: 31,
    fontsize: 21,
    img: 'https://avatars.githubusercontent.com/u/37553901?v=4',
);
```

### Tooltips without role

![image](https://user-images.githubusercontent.com/37553901/134119791-21eba1c7-e50f-44aa-b66d-30c41de98a87.png)

``` dart
ProfilePicture(
    name: 'Aditya Dharmawan Saputra',
    role: '',
    radius: 31,
    fontsize: 21,
    tooltip: true,
);
```

### Tooltips with role

![image](https://user-images.githubusercontent.com/37553901/134119720-ec140db1-b3ab-4422-90af-323b02d5060d.png)

``` dart
ProfilePicture(
    name: 'Aditya Dharmawan Saputra',
    role: 'ADMINISTRATOR',
    radius: 31,
    fontsize: 21,
    tooltip: true,
);
```

### Tooltips with role and images

![image](https://user-images.githubusercontent.com/37553901/134119679-44c74bcb-b51b-463f-ade1-42697d68eac5.png)

``` dart
ProfilePicture(
    name: 'Aditya Dharmawan Saputra',
    role: 'ADMINISTRATOR',
    radius: 31,
    fontsize: 21,
    tooltip: true,
    img: 'https://avatars.githubusercontent.com/u/37553901?v=4',
);
```


CHANGELOG.md
# CHANGE LOG

## 1.0.0

* Initial Release with 8 feature.

## 1.0.1

* Update Readme.

## 1.0.2

* Update description.

## 2.0.0

* Migrated to null-safety
* Upgrade sdk to 2.12.0 as a minimum
* Fixed the compile issues where nullability is now a thing
* Fixing color name for the letter 'Z'
* Add horizontal and vertical scroll for demo apps (you can run demo apps inside example folder)
* Add automatic testing when contributor do pull request (github)
* Add auto build gh-pages for live demo (you can see live demo here <https://adityadees.github.io/flutter_profile_picture/>)


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:flutter_profile_picture/flutter_profile_picture.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'FLUTTER PROFILE PICTURE',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Example FPP'),
      ),
      body: SingleChildScrollView(
        scrollDirection: Axis.vertical,
        child: SingleChildScrollView(
          scrollDirection: Axis.horizontal,
          child: Container(
            width: 500,
            padding: EdgeInsets.symmetric(horizontal: 10, vertical: 10),
            child: Table(
              border: TableBorder.all(),
              columnWidths: const <int, TableColumnWidth>{
                0: FlexColumnWidth(),
                1: FlexColumnWidth(),
                2: FlexColumnWidth(),
              },
              defaultVerticalAlignment: TableCellVerticalAlignment.middle,
              children: <TableRow>[
                _exampleDefault(),
                _exampleRandomColor(),
                _exampleMaxLetter(),
                _exampleEmptyName(),
                _exampleImages(),
                _exampleTooltipsWithoutRole(),
                _exampleTooltipsWithRole(),
                _exampleTooltipsWithRoleAndImages(),
              ],
            ),
          ),
        ),
      ),
    );
  }

  TableRow _exampleTooltipsWithRoleAndImages() {
    return TableRow(
      children: <Widget>[
        Container(
          padding: EdgeInsets.symmetric(horizontal: 10, vertical: 10),
          child: Text(
            '8. Tooltips with role and images',
            style: TextStyle(
              fontSize: 14,
              fontWeight: FontWeight.bold,
            ),
          ),
        ),
        Container(
          padding: EdgeInsets.symmetric(horizontal: 10, vertical: 10),
          // enable tooltip feature with role and image
          // same as example 7
          // but you can add parameter img = 'your_img'
          // background color and initial name will be replaced with the image
          child: ProfilePicture(
            name: 'Aditya Dharmawan Saputra',
            role: 'ADMINISTRATOR',
            radius: 31,
            fontsize: 21,
            tooltip: true,
            img: 'https://avatars.githubusercontent.com/u/37553901?v=4',
          ),
        ),
        Center(
          child: Container(
            padding: EdgeInsets.symmetric(horizontal: 10, vertical: 10),
            child: Text(
              'Replace name with image, you can click the images if you want to show tooltips with full name and role',
              style: TextStyle(
                fontSize: 12,
              ),
            ),
          ),
        ),
      ],
    );
  }

  TableRow _exampleTooltipsWithRole() {
    return TableRow(
      children: <Widget>[
        Container(
          padding: EdgeInsets.symmetric(horizontal: 10, vertical: 10),
          child: Text(
            '7. Tooltips with role',
            style: TextStyle(
              fontSize: 14,
              fontWeight: FontWeight.bold,
            ),
          ),
        ),
        Container(
          padding: EdgeInsets.symmetric(horizontal: 10, vertical: 10),
          // enable tooltip feature with role
          // same as example 6
          // but you can add parameter role = 'your_role'
          // role will be appear under name
          child: ProfilePicture(
            name: 'Aditya Dharmawan Saputra',
            role: 'ADMINISTRATOR',
            radius: 31,
            fontsize: 21,
            tooltip: true,
          ),
        ),
        Center(
          child: Container(
            padding: EdgeInsets.symmetric(horizontal: 10, vertical: 10),
            child: Text(
              'Click to show tooltips with full name and role',
              style: TextStyle(
                fontSize: 12,
              ),
            ),
          ),
        ),
      ],
    );
  }

  TableRow _exampleTooltipsWithoutRole() {
    return TableRow(
      children: <Widget>[
        Container(
          padding: EdgeInsets.symmetric(horizontal: 10, vertical: 10),
          child: Text(
            '6. Tooltips without role',
            style: TextStyle(
              fontSize: 14,
              fontWeight: FontWeight.bold,
            ),
          ),
        ),
        Container(
          padding: EdgeInsets.symmetric(horizontal: 10, vertical: 10),
          // enable tooltip feature
          // set tooltip = true
          // default is false
          // when you click the picture
          // name will be appear
          child: ProfilePicture(
            name: 'Aditya Dharmawan Saputra',
            role: '',
            radius: 31,
            fontsize: 21,
            tooltip: true,
          ),
        ),
        Center(
          child: Container(
            padding: EdgeInsets.symmetric(horizontal: 10, vertical: 10),
            child: Text(
              'Click to show tooltips with full name',
              style: TextStyle(
                fontSize: 12,
              ),
            ),
          ),
        ),
      ],
    );
  }

  TableRow _exampleImages() {
    return TableRow(
      children: <Widget>[
        Container(
          padding: EdgeInsets.symmetric(horizontal: 10, vertical: 10),
          child: Text(
            '5. Images',
            style: TextStyle(
              fontSize: 14,
              fontWeight: FontWeight.bold,
            ),
          ),
        ),
        Container(
          padding: EdgeInsets.symmetric(horizontal: 10, vertical: 10),
          // change background color to image
          // set img = 'your_url_img'
          child: ProfilePicture(
            name: 'Aditya Dharmawan Saputra',
            radius: 31,
            fontsize: 21,
            img: 'https://avatars.githubusercontent.com/u/37553901?v=4',
          ),
        ),
        Center(
          child: Container(
            padding: EdgeInsets.symmetric(horizontal: 10, vertical: 10),
            child: Text(
              'add "img:your_url" for replace the name with images',
              style: TextStyle(
                fontSize: 12,
              ),
            ),
          ),
        ),
      ],
    );
  }

  TableRow _exampleEmptyName() {
    return TableRow(
      children: <Widget>[
        Container(
          padding: EdgeInsets.symmetric(horizontal: 10, vertical: 10),
          child: Text(
            '4. Empty Name',
            style: TextStyle(
              fontSize: 14,
              fontWeight: FontWeight.bold,
            ),
          ),
        ),
        Container(
          padding: EdgeInsets.symmetric(horizontal: 10, vertical: 10),
          // for handling user without name
          // set name = '' (empty string)
          // will set default background color without name
          child: ProfilePicture(
            name: '',
            radius: 31,
            fontsize: 21,
          ),
        ),
        Center(
          child: Container(
            padding: EdgeInsets.symmetric(horizontal: 10, vertical: 10),
            child: Text(
              'You can make a perfect circle without filling the name',
              style: TextStyle(
                fontSize: 12,
              ),
            ),
          ),
        ),
      ],
    );
  }

  TableRow _exampleMaxLetter() {
    return TableRow(
      children: <Widget>[
        Container(
          padding: EdgeInsets.symmetric(horizontal: 10, vertical: 10),
          child: Text(
            '3. Max Letter',
            style: TextStyle(
              fontSize: 14,
              fontWeight: FontWeight.bold,
            ),
          ),
        ),
        Container(
          padding: EdgeInsets.symmetric(horizontal: 10, vertical: 10),
          // set limit letter
          // use count = 'your_limit'
          // default is 2 if more than one words
          // default is 1 if less than two words
          child: ProfilePicture(
            name: 'Aditya Dharmawan Saputra',
            radius: 31,
            fontsize: 21,
            count: 3,
          ),
        ),
        Center(
          child: Container(
            padding: EdgeInsets.symmetric(horizontal: 10, vertical: 10),
            child: Text(
              'add "count:your_length" for set how much the letters shown on circle',
              style: TextStyle(
                fontSize: 12,
              ),
            ),
          ),
        ),
      ],
    );
  }

  TableRow _exampleRandomColor() {
    return TableRow(
      children: <Widget>[
        Container(
          padding: EdgeInsets.symmetric(horizontal: 10, vertical: 10),
          child: Text(
            '2. Random Color',
            style: TextStyle(
              fontSize: 14,
              fontWeight: FontWeight.bold,
            ),
          ),
        ),
        Container(
          padding: EdgeInsets.symmetric(horizontal: 10, vertical: 10),
          // set random = true
          // default is false
          child: ProfilePicture(
            name: 'Aditya Dharmawan Saputra',
            radius: 31,
            fontsize: 21,
            random: true,
          ),
        ),
        Center(
          child: Container(
            padding: EdgeInsets.symmetric(horizontal: 10, vertical: 10),
            child: Text(
              'add random:true if you want to give random color',
              style: TextStyle(
                fontSize: 12,
              ),
            ),
          ),
        ),
      ],
    );
  }

  TableRow _exampleDefault() {
    return TableRow(
      children: <Widget>[
        Container(
          padding: EdgeInsets.symmetric(horizontal: 10, vertical: 10),
          child: Text(
            '1. Default',
            style: TextStyle(
              fontSize: 14,
              fontWeight: FontWeight.bold,
            ),
          ),
        ),
        Center(
          child: Container(
            child: Row(
              children: [
                // if there is only one word.
                ProfilePicture(
                  name: 'Dees',
                  radius: 31,
                  fontsize: 21,
                ),
                SizedBox(width: 5),
                // if more than one word.
                ProfilePicture(
                  name: 'Aditya Dharmawan Saputra',
                  radius: 31,
                  fontsize: 21,
                ),
              ],
            ),
          ),
        ),
        Container(
          padding: EdgeInsets.symmetric(horizontal: 10, vertical: 10),
          child: Text(
            'This is default style. Default color based on First Initial Name',
            style: TextStyle(
              fontSize: 12,
            ),
          ),
        ),
      ],
    );
  }
}
